### DISCUSSION TITLE
Suggestion: Give a sponsor tier access to more than one private repository.

### DISCUSSION THREAD

**Discussion by LlmDl** [community/community#17986] [Other Feature Feedback, Questions, & Ideas]
It is currently possible to give a sponsorship tier access to a repository, but only one repository. This is a really nice addition.

I am hoping that soon it will be possible for a tier to have access granted to multiple private repos (I want to give my sponsors access to 4+ repos.)

Thanks for reading my suggestion!

**Comment by LlmDl** [comment]
Giving this suggestion a 6-month bump. Please consider making it possible for sponsor tiers to receive access to more than one repository!

**Reply by nghiemnguyen23** [reply]
Nghiem24

**Comment by EduApps-CDG** [comment]
It would be great if GitHub Sponsors could allow sponsors to support multiple private repositories in a single tier. This would not only provide more flexibility for sponsors to choose which repositories they want to support, but it would also benefit developers by allowing them to offer more value to their sponsors. I believe that this feature would enhance the overall experience of GitHub Sponsors and encourage more sponsors to support their favorite open-source projects.

**Comment by LlmDl** [comment]
Giving this suggestion a 12-month bump. Please consider making it possible for sponsor tiers to receive access to more than one repository!

**Comment by Chizaruu** [comment]
Reiterating the importance of comments over mere reactions, I'm giving this another bump.

`Access to a private Sponsor Repository` -> `Access to private Sponsor Repositories`

**Comment by dougmercer** [comment]
As an educational YouTuber, I create videos about Python. I create a repository that contains the code used in each video I create.

I'd like to grant my sponsors access to these repositories. With the current limitations, I would need to use a mono-repo rather than multiple per-video repositories. Being able to grant access to multiple videos would improve the sponsorship experience for my sponsors.

**Comment by martylamb** [comment]
+1 multiple private repositories associated with a tier would be extremely helpful

**Comment by robcaulk** [comment]
+1 bump for all of the same reasons already iterated in this thread.

**Comment by razonyang** [comment]
Please considering adding this feature, I don't want to put all _private_ things (articles, softwares and so on) in single one mono-repo, since it's hard to maintain.

**Comment by ricaun** [comment]
Hoo noo! I discovered today that I can only add one repo... 😐

**Comment by birkjernstrom** [comment]
We're developing polar.sh (https://github.com/polarsource/polar) as a creator platform for developers – built open source on GitHub.

Today, we introduced the ability to upsell multiple private GitHub repositories across the subscription tiers you could setup with Polar. Combined with other benefits: Premium posts, newsletter, Discord invites & automation of syncing sponsor logotypes across your README, site, docs & Polar posts.

We're growing & shipping quickly based on feedback from developers. We want to empower a new generation of independent developers pursuing their passions full-time. Join our Discord and let us know how we can help 🙂

**Reply by razonyang** [reply]
Hi, I'm interested in it, but I think it seems the following features will be lost:

- Sponsors won't get GitHub _sponsor_ badge on their GitHub profile.
- Unable to indicate whether if a user is a sponsor on GitHub, for example, a sponsor creates an issue, will have a _sponsor_ label (or sth else) next to their avatar and name.

Please correct me if I'm wrong.

**Reply by birkjernstrom** [reply]
Sorry for the delay @razonyang. Yes, both of those assumptions are correct. However, we have ideas for both in our backlog so it's something we hope to solve in the near future.

**Comment by dcblogdev** [comment]
I also would like to give sponsors access to multiple repositories, each repo is a separate codebase.

**Comment by quimodotcom** [comment]
Bump also

**Comment by gbtilabs** [comment]
We're considering GitHub Sponsors and GitHub Oauth as a licensing system as well as a way to further gate our open source content to our community. 10 tiers is good to start, but we'd rather provide access to multiple repos under one tier, rather than create many tiers for our different oss products.

**Comment by Vercidium** [comment]
Would love to be able to grant sponsors access to multiple repositories

**Comment by ve3** [comment]
Hello from 2026.
I also request for this too.

**Comment by kennytv** [comment]
I too would love this, as of now I have to manually add members to a secondary repository and keep track of who to remove again...
